src/github.com/matrix-org/dendrite/federationapi/writers/threepid.go

```diff
@@ -120,11 +121,30 @@ func createInviteFrom3PIDInvite(
 	}
 
 	if !queryRes.RoomExists {
-		// TODO: Use federation to auth the event
-		return nil, nil
+		// Use federation to auth the event
+		var remoteServers []gomatrixserverlib.ServerName
+		_, remoteServers[0], err = gomatrixserverlib.SplitID('@', inv.Sender)
+		if err != nil {
+			return nil, err
+		}
+		// Fallback to the room's server if the sender's domain is the same as
+		// the current server's
+		_, remoteServers[1], err = gomatrixserverlib.SplitID('!', inv.RoomID)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, server := range remoteServers {
+			if server == cfg.Matrix.ServerName {
+				continue
+			}
+
+			err = federation.ExchangeThirdPartyInvite(server, *builder)
```

Comment:
If there is an error you may want to continue and try the next server.

Reply:
Fixed in https://github.com/matrix-org/dendrite/pull/222/commits/e1c8861c4ab96463fcc1aa1c759544e48aae7adc